lib/matplotlib/tests/test_legend.py

```diff
@@ -1429,31 +1429,16 @@ def test_legend_text():
     assert_allclose(leg_bboxes[1].bounds, leg_bboxes[0].bounds)
 
 
-def test_boxplot_legend():
-    # Test that boxplot legends handles are patches
-    # and labels are generated from boxplot's labels parameter.
-    fig, axs = plt.subplots()
-    A = 5*np.random.rand(100, 1)
-    B = 10*np.random.rand(100, 1) - 5
-    C = 7*np.random.rand(100, 1) - 5
-    labels = ['a', 'b', 'c']
-
-    bp0 = axs.boxplot(A, positions=[0], patch_artist=True, labels=labels[0])
-    bp1 = axs.boxplot(B, positions=[1], patch_artist=True, labels=labels[1])
-    bp2 = axs.boxplot(C, positions=[2], patch_artist=True, labels=labels[2])
-    # red, blue, green
-    colors = [(1.0, 0.0, 0.0, 1), (0.0, 0.0, 1.0, 1), (0.0, 0.5, 0.0, 1)]
-    box_list = [bp0, bp1, bp2]
-    # Set colors to the boxes
-    lbl_index = 0
-    for b_plot, color in zip(box_list, colors):
-        for patch in b_plot['boxes']:
-            patch.set_color(color)
-            lbl_index += 1
-
-    legend = axs.legend()
-    for index, handle in enumerate(legend.legend_handles):
-        assert isinstance(handle, mpl.patches.Rectangle)
-        assert handle.get_facecolor() == colors[index]
-        assert handle.get_edgecolor() == colors[index]
-        assert handle.get_label() == labels[index]
+def test_boxplot_labels():
+    # Test that boxplot(..., labels=) sets the tick labels but not legend entries
+    # This is not consistent with other plot types but is the current behavior.
+    fig, ax = plt.subplots()
+    np.random.seed(19680801)
+    data = np.random.random((10, 3))
+    bp = ax.boxplot(data, labels=['A', 'B', 'C'])
+    # Check that labels set the tick labels ...
+    assert [l.get_text() for l in ax.get_xticklabels()] == ['A', 'B', 'C']
+    # ... but not legend entries
+    handles, labels = ax.get_legend_handles_labels()
```

Comment:
Doesn't this need an `ax.legend()` call first to properley test if any legend entries are added? It seems to work as intended, but locally I get a `UserWarning: No artists with labels found to put in legend.  Note that artists whose label start with an underscore are ignored when legend() is called with no argument.` which is expected and I think should be caught in this test when `ax.legend()` is called.

Reply:
`get_legend_handles_and_labels()` is the "give me all entries for a legend" function. It traverses all artists and return handle+label for those that should be included in a legend because they have a label.

`ax.legend()` calls that internally. IMHO it's more straight forward to just check that there are no artists for a legend that to try and create a legend and check that this warns that there are no entries.